Model/Admin/Notification/Messages.php

```diff
@@ -0,0 +1,127 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2017 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Model\Admin\Notification;
+
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Nosto\Tagging\Helper\Scope as NostoHelperScope;
+use Magento\Framework\Notification\MessageInterface;
+
+class Messages implements MessageInterface
+{
+    private $nostoHelperScope;
+    private $nostoHelperAccount;
+    private $message;
+    private $storeNames;
+    private $display;
+
+    /**
+     * Messages constructor.
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoHelperScope $nostoHelperScope
+     */
+    public function __construct(
+        NostoHelperAccount $nostoHelperAccount,
+        NostoHelperScope $nostoHelperScope
+    ) {
+        $this->nostoHelperScope = $nostoHelperScope;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+    }
+
+    /**
+     * @return \Magento\Framework\Phrase|string
+     */
+    public function getText()
+    {
+        return __($this->message);
+    }
+
+    /**
+     * @return string
+     */
+    public function getIdentity()
+    {
+        return 'Nosto_Account_Notification';
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDisplayed()
+    {
+        $stores = $this->nostoHelperScope->getStores();
+
+        foreach ($stores as $store) {
+            //Check if the store is connected to Nosto
+            if ($this->nostoHelperAccount->findAccount($store)
+                && $this->nostoHelperAccount->isDomainValid($store) === false) {
+                    $this->storeNames[] = $store->getName();
```

Comment:
Is there a reason why `storeNames` is a class variable? Could it just be a local variable that you'd pass to `buildMessage`?